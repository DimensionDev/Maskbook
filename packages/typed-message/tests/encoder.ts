/* eslint-disable @typescript-eslint/await-thenable */
import { test, expect } from '@jest/globals'
import {
    makeTypedMessageText,
    encodeTypedMessageToDocument,
    makeTypedMessageSerializableTupleFromList,
    decodeTypedMessageFromDocument,
} from '../dist/base/index.js'

const meta = new Map<string, any>([
    ['com.example.test', 'hi'],
    ['com.example.test2', { a: 1, b: [2], c: new Uint8Array([1, 2, 3, 4]) }],
])

test('Serialize Text', () => {
    const result = encodeTypedMessageToDocument(makeTypedMessageText('Hello world', meta))
    expect(result).toMatchSnapshot()
    expect(decodeTypedMessageFromDocument(result)).toMatchSnapshot()
})

test('Serialize Tuple', () => {
    const msg = makeTypedMessageSerializableTupleFromList(
        makeTypedMessageText('Hello world', meta),
        makeTypedMessageText('another'),
    )
    const result = encodeTypedMessageToDocument(msg)
    expect(result).toMatchSnapshot()
    expect(decodeTypedMessageFromDocument(result)).toMatchSnapshot()
})
