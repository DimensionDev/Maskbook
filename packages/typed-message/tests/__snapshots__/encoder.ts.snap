// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Serialize Text 1`] = `Uint8Array [ 9300ab48656c6c6f20776f726c6482b0636f6d2e6578616d706c652e74657374a26869b1636f6d2e6578616d706c652e746573743283a16101a1629102a163c40401020304 ]`;

exports[`Serialize Text 2`] = `
Ok({
  "content": "Hello world",
  "meta": {
    "com.example.test": "hi",
    "com.example.test2": {
      "a": 1,
      "b": [
        2,
      ],
      "c": Uint8Array [ 01020304 ],
    },
  },
  "serializable": true,
  "type": "text",
  "version": 1,
})
`;

exports[`Serialize Tuple 1`] = `Uint8Array [ 95010001c09294010182b0636f6d2e6578616d706c652e74657374a26869b1636f6d2e6578616d706c652e746573743283a16101a1629102a163c40401020304ab48656c6c6f20776f726c64940101c0a7616e6f74686572 ]`;

exports[`Serialize Tuple 2`] = `
Ok({
  "items": [
    {
      "content": "Hello world",
      "meta": {
        "com.example.test": "hi",
        "com.example.test2": {
          "a": 1,
          "b": [
            2,
          ],
          "c": Uint8Array [ 01020304 ],
        },
      },
      "serializable": true,
      "type": "text",
      "version": 1,
    },
    {
      "content": "another",
      "meta": undefined,
      "serializable": true,
      "type": "text",
      "version": 1,
    },
  ],
  "meta": undefined,
  "type": "tuple",
})
`;
